docs/porting.md

```diff
@@ -22,17 +22,19 @@ Extension modules need to explicitly indicate they support running with the GIL
 disabled, otherwise a warning is printed and the GIL is re-enabled at runtime
 after importing a module that does not support the GIL. 
 
-!!! note
-
-    Currently it is not possible for extensions written in Cython to declare
-    they support running without the GIL. Work is under way to add support
-    (see [cython#6242](https://github.com/cython/cython/pull/6242)).
-
 C++ extension modules making use of `pybind11` can easily declare support for
 running with the GIL disabled via the
 [`gil_not_used`](https://pybind11.readthedocs.io/en/stable/reference.html#_CPPv4N7module_23create_extension_moduleEPKcPKcP10module_def16mod_gil_not_used)
 argument to `create_extension_module`.
 
+Starting with Cython 3.1.0 (only available via the nightly wheels or the `master`
+branch as of right now), extension modules written in Cython can also do so using the
+[`freethreading_compatible`](https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#compiler-directives)
+compiler directive. It can be enabled either per module as a directive
+(`# cython: freethreading_compatible=True`) in `.pyx` files, or globally by adding
+`-Xfreethreading_compatible=True` to the Cython arguments via the project's
+build system.
+
```

Comment:
Here's some example code to do this via build configuration with setuptools and meson:

```python
compiler_directives = {'freethreading_compatible': True}

setup(
    ext_modules=cythonize(
        extensions,
        compiler_directives=compiler_directives)
)
```

```meson
cython_args = []
if cy.version().version_compare('>=3.1.0')
  cython_args += ['-Xfreethreading_compatible=True']
endif
```

Meson compiler versions don't include alpha or beta qualifiers, so no need to check that for cython.

For the setuptools one, you could note that you can use e.g. `packaging.version` to check for a minimum Cython version.

Reply:
I like the idea of including example build config code - that'll definitely be helpful for users.

The Meson code should include an `extension_module` call as well I think, something like:
```meson
py.extension_module('modulename'
    'source.pyx',
    cython_args: cython_args,
    ...
)
```

Even nicer if we had a CMake/scikit-build-core version of it as well. Maybe @henryiii could suggest an idiomatic snippet?

Reply:
Thanks both for the suggestions! I added setuptools and Meson examples, but I don't know enough about CMake/scikit-build-core to do that as well.

Reply:
This will be better once we finish https://github.com/scikit-build/cython-cmake/pull/19. Will try to do that this week and then update here.